apps/wing/src/commands/compile.ts

```diff
@@ -280,6 +284,28 @@ export async function compile(entrypoint: string, options: CompileOptions): Prom
   return synthDir;
 }
 
+function annotatePreflightError(e: any): any {
+  if (!(e instanceof Error)) {
+    return e;
+  }
+
+  if (e.message.startsWith("There is already a Construct with name")) {
```

Comment:
Yak

I was wondering if perhaps we can deal with this at the compiler level but that's good enough for now...

Reply:
Curious what you had in mind?

I couldn't think of any solutions based in wingc that would hold water after you start introducing dynamic IDs:

```
let x = 1;
new cloud.Bucket() as "bucket-${x}";
new cloud.Bucket() as "bucket-${x}";
```

and/or dynamic scopes:

```
new cloud.Bucket() in scope1;
new cloud.Bucket() as scope2;
```

There might be heuristics that catch some cases, but I think the risk is that having errors that only _sometimes_ appear as red underlines in VS Code would feel bad to use as a developer.

One idea I had for surfacing errors better was that if you haven't edited your code for 500ms or so, the VS Code extension might be able to compile your app to the sim target in the background and then highlight any preflight errors in your editor, but I think that might require source maps.

Reply:
I'll mark this conversation as resolved, but feel free to reopen or add a comment on the issue anything else comes to mind